File: two-sum2.py
class Solution:
    def two_sum(self, nums:list[int], target:int)->list[int]:
        left = 0
        right = len(nums) - 1
        ret = [0,0]

        while( left < right ):
            if target == nums[left] + nums[right]:
                ret = [left+1, right+1]
                break

            if target > nums[left] + nums[right]:
                left += 1
            else:
                right -= 1

        return ret

numbers = [2,7,11,15]
target = 9
sol = Solution()
ans = sol.two_sum(numbers, target)
print(ans)

File: two-sum2.php
<?php
/**
*
* LeetCode 167. Two Sum II - Input Array Is Sorted using Binary Sort PHP 
*
**/

class Solution{
    function twoSum($nums, $target){
        $left = 0;
        $right = count($nums)-1;

        while( $left < $right ){
            if( $target == $nums[$left] + $nums[$right] ){
                return [$left+1, $right +1];
            }

            if( $target > $nums[$left] + $nums[$right]){
                $left++;
            }
            else {
                $right--;
            }
        }
        return [0,0];
    }
}

$numbers = [2,7,11,15];
$target = 9;

// $numbers = [2,3,4];
// $target = 6;

$sol = new Solution();
$ans = $sol->twosum($numbers, $target);
print_r($ans);
